{
 "cells": [
  {
   "metadata": {},
   "cell_type": "markdown",
   "source": "Script to do analysis on the integrated data (FAIRE + RNA-seq), chi-squared test. Change the overlap and min_reps parameters to get the combination that you want. Default here: overlap = '0.9', minreps = 1 ",
   "id": "f9c1542aaf0ecfea"
  },
  {
   "metadata": {
    "ExecuteTime": {
     "end_time": "2024-05-27T14:35:12.598180Z",
     "start_time": "2024-05-27T14:35:12.595938Z"
    }
   },
   "cell_type": "code",
   "source": [
    "import os\n",
    "import pandas as pd\n",
    "import numpy as np\n",
    "from scipy.stats import chi2_contingency"
   ],
   "id": "bd566cb807a71c2d",
   "outputs": [],
   "execution_count": 5
  },
  {
   "metadata": {
    "ExecuteTime": {
     "end_time": "2024-05-27T14:35:13.297813Z",
     "start_time": "2024-05-27T14:35:13.295587Z"
    }
   },
   "cell_type": "code",
   "source": [
    "# minimum number of peaks in each condition (here male/partheno)\n",
    "min_reps = '1'\n",
    "# overlap used with bedtools intersect (-F parameter)\n",
    "overlap = '0.9'  # max = 1.0"
   ],
   "id": "31c0bf29cdf0ba4f",
   "outputs": [],
   "execution_count": 6
  },
  {
   "metadata": {
    "ExecuteTime": {
     "end_time": "2024-05-27T14:35:14.009911Z",
     "start_time": "2024-05-27T14:35:14.007286Z"
    }
   },
   "cell_type": "code",
   "source": [
    "cwd = os.getcwd() # current working directory. All scripts in this directory. Inside : RNA, FAIRE, and integrated\n",
    "RNA_directory = cwd + '/RNA/' # there's also the R scripts and result of the AskoR analysis inside\n",
    "FAIRE_directory = cwd + '/FAIRE/'\n",
    "integrated_directory = cwd + '/integrated/'"
   ],
   "id": "e0e73aa3bf49cee1",
   "outputs": [],
   "execution_count": 7
  },
  {
   "metadata": {
    "ExecuteTime": {
     "end_time": "2024-05-27T14:35:14.564074Z",
     "start_time": "2024-05-27T14:35:14.560756Z"
    }
   },
   "cell_type": "code",
   "source": [
    "# Prepare all headers for the file imports\n",
    "header_DAR_broad = ['chr', 'start', 'end', 'interval_id', 'num_peaks', 'num_samples', 'Male1_bool', 'Male2_bool', 'Partheno1_bool', 'Partheno2_bool', 'Partheno3_bool', 'Male1_fc', 'Male2_fc', 'Partheno1_fc', 'Partheno2_fc', 'Partheno3_fc', 'Male1_qval', 'Male2_qval', 'Partheno1_qval', 'Partheno2_qval', 'Partheno3_qval', 'Male1_pval', 'Male2_pval', 'Partheno1_pval', 'Partheno2_pval', 'Partheno3_pval', 'Male1_start', 'Male2_start', 'Partheno1_start', 'Partheno2_start', 'Partheno3_start', 'Male1_end', 'Male2_end', 'Partheno1_end', 'Partheno2_end', 'Partheno3_end']\n",
    "header_DAR_narrow = header_DAR_broad + ['Male1_summit', 'Male2_summit', 'Partheno1_summit', 'Partheno2_summit', 'Partheno3_summit']\n",
    "header_DEG_gff = ['seqid', 'source', 'type', 'DEG_start', 'DEG_end', 'score', 'strand', 'phase', 'attributes']\n",
    "header_DEG_bed = ['seqid', 'DEG_start', 'DEG_end', 'strand', 'gene']\n",
    "header_DAR_broad_DEG = header_DAR_broad[:4] + ['strand', 'gene'] + header_DAR_broad[4:]\n",
    "header_DAR_narrow_DEG = header_DAR_narrow[:4] + ['strand', 'gene'] + header_DAR_narrow[4:]"
   ],
   "id": "9bfa0456bc18f58a",
   "outputs": [],
   "execution_count": 8
  },
  {
   "metadata": {
    "ExecuteTime": {
     "end_time": "2024-05-27T14:35:15.540459Z",
     "start_time": "2024-05-27T14:35:15.202308Z"
    }
   },
   "cell_type": "code",
   "source": [
    "# DAR broad\n",
    "DAR_broad_male_path = f'{FAIRE_directory}DAR_broad_male_minreps{min_reps}.bed'\n",
    "DAR_broad_partheno_path = f'{FAIRE_directory}DAR_broad_partheno_minreps{min_reps}.bed'\n",
    "intergroup_broad_path = f'{FAIRE_directory}DAR_broad_intergroup_minreps{min_reps}.bed'\n",
    "DAR_broad_male = pd.read_csv(DAR_broad_male_path, sep='\\t', names=header_DAR_broad)\n",
    "DAR_broad_partheno = pd.read_csv(DAR_broad_partheno_path, sep='\\t', names=header_DAR_broad)\n",
    "intergroup_broad = pd.read_csv(intergroup_broad_path, sep='\\t', names=header_DAR_broad)\n",
    "\n",
    "# DAR narrow\n",
    "DAR_narrow_male_path = f'{FAIRE_directory}DAR_narrow_male_minreps{min_reps}.bed'\n",
    "DAR_narrow_partheno_path = f'{FAIRE_directory}DAR_narrow_partheno_minreps{min_reps}.bed'\n",
    "intergroup_narrow_path = f'{FAIRE_directory}DAR_narrow_intergroup_minreps{min_reps}.bed'\n",
    "DAR_narrow_male = pd.read_csv(DAR_narrow_male_path, sep='\\t', names=header_DAR_narrow)\n",
    "DAR_narrow_partheno = pd.read_csv(DAR_narrow_partheno_path, sep='\\t', names=header_DAR_narrow)\n",
    "intergroup_narrow = pd.read_csv(intergroup_narrow_path, sep='\\t', names=header_DAR_narrow)\n",
    "\n",
    "DAR_broad_male"
   ],
   "id": "48dd501424ca9947",
   "outputs": [
    {
     "data": {
      "text/plain": [
       "                chr    start      end     interval_id  num_peaks  num_samples  \\\n",
       "0        scaffold_1    43510    43769      Interval_2          2            2   \n",
       "1        scaffold_1   205473   205653      Interval_8          1            1   \n",
       "2        scaffold_1   666042   666280     Interval_25          1            1   \n",
       "3        scaffold_1   856129   856490     Interval_37          2            2   \n",
       "4        scaffold_1  1054779  1055762     Interval_46          1            1   \n",
       "...             ...      ...      ...             ...        ...          ...   \n",
       "14502  scaffold_553     3414     3767  Interval_44207          1            1   \n",
       "14503   scaffold_75    21132    21352  Interval_44231          1            1   \n",
       "14504   scaffold_81      300      688  Interval_44238          1            1   \n",
       "14505   scaffold_88    18958    19913  Interval_44245          2            2   \n",
       "14506    scaffold_9    89987    90307  Interval_44249          1            1   \n",
       "\n",
       "       Male1_bool  Male2_bool  Partheno1_bool  Partheno2_bool  ...  \\\n",
       "0            True        True           False           False  ...   \n",
       "1            True       False           False           False  ...   \n",
       "2           False        True           False           False  ...   \n",
       "3            True        True           False           False  ...   \n",
       "4           False        True           False           False  ...   \n",
       "...           ...         ...             ...             ...  ...   \n",
       "14502       False        True           False           False  ...   \n",
       "14503       False        True           False           False  ...   \n",
       "14504       False        True           False           False  ...   \n",
       "14505        True        True           False           False  ...   \n",
       "14506       False        True           False           False  ...   \n",
       "\n",
       "       Male1_start Male2_start  Partheno1_start  Partheno2_start  \\\n",
       "0            43510     43516.0              NaN              NaN   \n",
       "1           205473         NaN              NaN              NaN   \n",
       "2              NaN    666042.0              NaN              NaN   \n",
       "3           856129    856200.0              NaN              NaN   \n",
       "4              NaN   1054779.0              NaN              NaN   \n",
       "...            ...         ...              ...              ...   \n",
       "14502          NaN      3414.0              NaN              NaN   \n",
       "14503          NaN     21132.0              NaN              NaN   \n",
       "14504          NaN       300.0              NaN              NaN   \n",
       "14505        19225     18958.0              NaN              NaN   \n",
       "14506          NaN     89987.0              NaN              NaN   \n",
       "\n",
       "       Partheno3_start  Male1_end  Male2_end  Partheno1_end  Partheno2_end  \\\n",
       "0                  NaN      43698    43769.0            NaN            NaN   \n",
       "1                  NaN     205653        NaN            NaN            NaN   \n",
       "2                  NaN        NaN   666280.0            NaN            NaN   \n",
       "3                  NaN     856490   856409.0            NaN            NaN   \n",
       "4                  NaN        NaN  1055762.0            NaN            NaN   \n",
       "...                ...        ...        ...            ...            ...   \n",
       "14502              NaN        NaN     3767.0            NaN            NaN   \n",
       "14503              NaN        NaN    21352.0            NaN            NaN   \n",
       "14504              NaN        NaN      688.0            NaN            NaN   \n",
       "14505              NaN      19454    19913.0            NaN            NaN   \n",
       "14506              NaN        NaN    90307.0            NaN            NaN   \n",
       "\n",
       "       Partheno3_end  \n",
       "0                NaN  \n",
       "1                NaN  \n",
       "2                NaN  \n",
       "3                NaN  \n",
       "4                NaN  \n",
       "...              ...  \n",
       "14502            NaN  \n",
       "14503            NaN  \n",
       "14504            NaN  \n",
       "14505            NaN  \n",
       "14506            NaN  \n",
       "\n",
       "[14507 rows x 36 columns]"
      ],
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>chr</th>\n",
       "      <th>start</th>\n",
       "      <th>end</th>\n",
       "      <th>interval_id</th>\n",
       "      <th>num_peaks</th>\n",
       "      <th>num_samples</th>\n",
       "      <th>Male1_bool</th>\n",
       "      <th>Male2_bool</th>\n",
       "      <th>Partheno1_bool</th>\n",
       "      <th>Partheno2_bool</th>\n",
       "      <th>...</th>\n",
       "      <th>Male1_start</th>\n",
       "      <th>Male2_start</th>\n",
       "      <th>Partheno1_start</th>\n",
       "      <th>Partheno2_start</th>\n",
       "      <th>Partheno3_start</th>\n",
       "      <th>Male1_end</th>\n",
       "      <th>Male2_end</th>\n",
       "      <th>Partheno1_end</th>\n",
       "      <th>Partheno2_end</th>\n",
       "      <th>Partheno3_end</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>scaffold_1</td>\n",
       "      <td>43510</td>\n",
       "      <td>43769</td>\n",
       "      <td>Interval_2</td>\n",
       "      <td>2</td>\n",
       "      <td>2</td>\n",
       "      <td>True</td>\n",
       "      <td>True</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>...</td>\n",
       "      <td>43510</td>\n",
       "      <td>43516.0</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>43698</td>\n",
       "      <td>43769.0</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>scaffold_1</td>\n",
       "      <td>205473</td>\n",
       "      <td>205653</td>\n",
       "      <td>Interval_8</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>True</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>...</td>\n",
       "      <td>205473</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>205653</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>scaffold_1</td>\n",
       "      <td>666042</td>\n",
       "      <td>666280</td>\n",
       "      <td>Interval_25</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>False</td>\n",
       "      <td>True</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>666042.0</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>666280.0</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>scaffold_1</td>\n",
       "      <td>856129</td>\n",
       "      <td>856490</td>\n",
       "      <td>Interval_37</td>\n",
       "      <td>2</td>\n",
       "      <td>2</td>\n",
       "      <td>True</td>\n",
       "      <td>True</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>...</td>\n",
       "      <td>856129</td>\n",
       "      <td>856200.0</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>856490</td>\n",
       "      <td>856409.0</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>scaffold_1</td>\n",
       "      <td>1054779</td>\n",
       "      <td>1055762</td>\n",
       "      <td>Interval_46</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>False</td>\n",
       "      <td>True</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>1054779.0</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>1055762.0</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>14502</th>\n",
       "      <td>scaffold_553</td>\n",
       "      <td>3414</td>\n",
       "      <td>3767</td>\n",
       "      <td>Interval_44207</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>False</td>\n",
       "      <td>True</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>3414.0</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>3767.0</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>14503</th>\n",
       "      <td>scaffold_75</td>\n",
       "      <td>21132</td>\n",
       "      <td>21352</td>\n",
       "      <td>Interval_44231</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>False</td>\n",
       "      <td>True</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>21132.0</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>21352.0</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>14504</th>\n",
       "      <td>scaffold_81</td>\n",
       "      <td>300</td>\n",
       "      <td>688</td>\n",
       "      <td>Interval_44238</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>False</td>\n",
       "      <td>True</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>300.0</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>688.0</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>14505</th>\n",
       "      <td>scaffold_88</td>\n",
       "      <td>18958</td>\n",
       "      <td>19913</td>\n",
       "      <td>Interval_44245</td>\n",
       "      <td>2</td>\n",
       "      <td>2</td>\n",
       "      <td>True</td>\n",
       "      <td>True</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>...</td>\n",
       "      <td>19225</td>\n",
       "      <td>18958.0</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>19454</td>\n",
       "      <td>19913.0</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>14506</th>\n",
       "      <td>scaffold_9</td>\n",
       "      <td>89987</td>\n",
       "      <td>90307</td>\n",
       "      <td>Interval_44249</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>False</td>\n",
       "      <td>True</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>89987.0</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>90307.0</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>14507 rows × 36 columns</p>\n",
       "</div>"
      ]
     },
     "execution_count": 9,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "execution_count": 9
  },
  {
   "metadata": {
    "ExecuteTime": {
     "end_time": "2024-05-27T14:35:16.416948Z",
     "start_time": "2024-05-27T14:35:16.401099Z"
    }
   },
   "cell_type": "code",
   "source": [
    "# DEG\n",
    "DEG_male_tss_bed = pd.read_csv(RNA_directory + 'DEG_male_tss.bed', sep='\\t', names=header_DEG_bed)\n",
    "DEG_partheno_tss_bed = pd.read_csv(RNA_directory + 'DEG_partheno_tss.bed', sep='\\t', names=header_DEG_bed)\n",
    "\n",
    "DEG_male_tss_bed"
   ],
   "id": "5fd782d4930e4e15",
   "outputs": [
    {
     "data": {
      "text/plain": [
       "             seqid  DEG_start  DEG_end strand    gene\n",
       "0       scaffold_1     182836   185836      +    g168\n",
       "1       scaffold_1     222426   225426      -    g169\n",
       "2       scaffold_1     223016   226016      +    g170\n",
       "3       scaffold_1     614736   617736      -    g190\n",
       "4       scaffold_1     614956   617956      +    g191\n",
       "...            ...        ...      ...    ...     ...\n",
       "2862  scaffold_418          0     1501      +     g33\n",
       "2863  scaffold_462       3046     6046      -  g31176\n",
       "2864  scaffold_488          0     1501      +  g20198\n",
       "2865  scaffold_512       1576     4576      -  g22836\n",
       "2866  scaffold_527       1816     4816      +  g22813\n",
       "\n",
       "[2867 rows x 5 columns]"
      ],
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>seqid</th>\n",
       "      <th>DEG_start</th>\n",
       "      <th>DEG_end</th>\n",
       "      <th>strand</th>\n",
       "      <th>gene</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>scaffold_1</td>\n",
       "      <td>182836</td>\n",
       "      <td>185836</td>\n",
       "      <td>+</td>\n",
       "      <td>g168</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>scaffold_1</td>\n",
       "      <td>222426</td>\n",
       "      <td>225426</td>\n",
       "      <td>-</td>\n",
       "      <td>g169</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>scaffold_1</td>\n",
       "      <td>223016</td>\n",
       "      <td>226016</td>\n",
       "      <td>+</td>\n",
       "      <td>g170</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>scaffold_1</td>\n",
       "      <td>614736</td>\n",
       "      <td>617736</td>\n",
       "      <td>-</td>\n",
       "      <td>g190</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>scaffold_1</td>\n",
       "      <td>614956</td>\n",
       "      <td>617956</td>\n",
       "      <td>+</td>\n",
       "      <td>g191</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2862</th>\n",
       "      <td>scaffold_418</td>\n",
       "      <td>0</td>\n",
       "      <td>1501</td>\n",
       "      <td>+</td>\n",
       "      <td>g33</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2863</th>\n",
       "      <td>scaffold_462</td>\n",
       "      <td>3046</td>\n",
       "      <td>6046</td>\n",
       "      <td>-</td>\n",
       "      <td>g31176</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2864</th>\n",
       "      <td>scaffold_488</td>\n",
       "      <td>0</td>\n",
       "      <td>1501</td>\n",
       "      <td>+</td>\n",
       "      <td>g20198</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2865</th>\n",
       "      <td>scaffold_512</td>\n",
       "      <td>1576</td>\n",
       "      <td>4576</td>\n",
       "      <td>-</td>\n",
       "      <td>g22836</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2866</th>\n",
       "      <td>scaffold_527</td>\n",
       "      <td>1816</td>\n",
       "      <td>4816</td>\n",
       "      <td>+</td>\n",
       "      <td>g22813</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>2867 rows × 5 columns</p>\n",
       "</div>"
      ]
     },
     "execution_count": 10,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "execution_count": 10
  },
  {
   "metadata": {
    "ExecuteTime": {
     "end_time": "2024-05-27T14:36:01.373056Z",
     "start_time": "2024-05-27T14:36:00.153913Z"
    }
   },
   "cell_type": "code",
   "source": [
    "# all genes X all expressed to get only all the expressed genes, but keeping the columns of the gff all genes table\n",
    "all_genes = pd.read_csv('Acyrthosiphon_pisum_JIC1_v1.0.scaffolds.braker2.gff', sep='\\t', names=header_DEG_gff)\n",
    "all_genes = all_genes[all_genes['type'] == 'gene']\n",
    "# Adding a column gene to the gff to make the join with the DEG lists easier\n",
    "all_genes['gene'] = all_genes['attributes'].str.extract(r'ID=(.*);')\n",
    "# Creating a new bed file with -/+ 1500pb around the TSS to define a larger promoter region\n",
    "all_genes_tss = pd.DataFrame({\n",
    "    'seqid': all_genes['seqid'],\n",
    "    'DEG_start': all_genes.apply(lambda x: x['DEG_start'] - 1500 if x['strand'] == '+' else x['DEG_end'] - 1500, axis=1).clip(lower=0),\n",
    "    'DEG_end': all_genes.apply(lambda x: x['DEG_start'] + 1500 if x['strand'] == '+' else x['DEG_end'] + 1500, axis=1).clip(lower=0),\n",
    "    'strand': all_genes['strand'],\n",
    "    'gene': all_genes['gene']\n",
    "})\n",
    "all_expressed_path = RNA_directory + 'AskoR/MaleVsPartheno/DEanalysis/DEtables/MalevsPartheno.txt'\n",
    "all_expressed = pd.read_csv(all_expressed_path, sep='\\t')\n",
    "all_expressed"
   ],
   "id": "8fd073abdeef1e3a",
   "outputs": [
    {
     "data": {
      "text/plain": [
       "         gene        contrast     Expression  Significance        PValue  \\\n",
       "0       g1040  MalevsPartheno  Male>Partheno             1  6.416811e-10   \n",
       "1       g1052  MalevsPartheno  Male>Partheno             1  3.633853e-10   \n",
       "2      g13894  MalevsPartheno  Male>Partheno             1  3.030168e-10   \n",
       "3      g13901  MalevsPartheno  Male>Partheno             1  3.285241e-10   \n",
       "4      g15240  MalevsPartheno  Male>Partheno             1  9.454212e-10   \n",
       "...       ...             ...            ...           ...           ...   \n",
       "14103   g9983  MalevsPartheno  Male=Partheno             0  9.732021e-01   \n",
       "14104   g9984  MalevsPartheno  Male=Partheno             0  9.981760e-01   \n",
       "14105   g9985  MalevsPartheno  Male=Partheno             0  7.550673e-01   \n",
       "14106   g9986  MalevsPartheno  Male=Partheno             0  6.892120e-01   \n",
       "14107   g9990  MalevsPartheno  Male=Partheno             0  9.798081e-01   \n",
       "\n",
       "           logFC           FC           FDR  Male_NormMeanCount  \\\n",
       "0      10.655157  1612.581797  3.717083e-07         1855.157830   \n",
       "1      10.739071  1709.158793  3.717083e-07        17499.549755   \n",
       "2      11.579582  3060.564620  3.717083e-07         4497.057113   \n",
       "3      10.223023  1195.188721  3.717083e-07         4413.650911   \n",
       "4      10.325433  1283.113518  3.717083e-07         1880.041721   \n",
       "...          ...          ...           ...                 ...   \n",
       "14103   0.338480     1.264423  9.999986e-01         1897.972198   \n",
       "14104   0.025302     1.017693  9.999986e-01         1913.176933   \n",
       "14105  -0.593703     1.509115  9.999986e-01          143.294916   \n",
       "14106   0.745883     1.677001  9.999986e-01         3310.595731   \n",
       "14107   0.191107     1.141640  9.999986e-01          422.372894   \n",
       "\n",
       "       Partheno_NormMeanCount  Male_CPMnormMeanCount  \\\n",
       "0                    1.028444             123.374449   \n",
       "1                   10.046700            1163.780935   \n",
       "2                    1.369622             299.069942   \n",
       "3                    3.534850             293.523140   \n",
       "4                    1.365800             125.029315   \n",
       "...                       ...                    ...   \n",
       "14103             1501.029867             126.221750   \n",
       "14104             1879.553424             127.232900   \n",
       "14105              217.053953               9.529610   \n",
       "14106             1974.267320             220.166100   \n",
       "14107              369.858517              28.089270   \n",
       "\n",
       "       Partheno_CPMnormMeanCount  \\\n",
       "0                       0.068395   \n",
       "1                       0.668141   \n",
       "2                       0.091085   \n",
       "3                       0.235080   \n",
       "4                       0.090830   \n",
       "...                          ...   \n",
       "14103                  99.823710   \n",
       "14104                 124.996800   \n",
       "14105                  14.434840   \n",
       "14106                 131.295600   \n",
       "14107                  24.596880   \n",
       "\n",
       "                                             Description  \n",
       "0                                  B-like cysteine ase 3  \n",
       "1                                  B-like cysteine ase 3  \n",
       "2                                SUMO- ligase PIAS2-like  \n",
       "3      -N-acetylglucosamine--peptide N-acetylglucosam...  \n",
       "4          facilitator superfamily domain-containing 6 B  \n",
       "...                                                  ...  \n",
       "14103                                                NaN  \n",
       "14104                             catecholamines up-like  \n",
       "14105                                        ST7 homolog  \n",
       "14106                                   pellucida domain  \n",
       "14107                                            : max X  \n",
       "\n",
       "[14108 rows x 13 columns]"
      ],
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>gene</th>\n",
       "      <th>contrast</th>\n",
       "      <th>Expression</th>\n",
       "      <th>Significance</th>\n",
       "      <th>PValue</th>\n",
       "      <th>logFC</th>\n",
       "      <th>FC</th>\n",
       "      <th>FDR</th>\n",
       "      <th>Male_NormMeanCount</th>\n",
       "      <th>Partheno_NormMeanCount</th>\n",
       "      <th>Male_CPMnormMeanCount</th>\n",
       "      <th>Partheno_CPMnormMeanCount</th>\n",
       "      <th>Description</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>g1040</td>\n",
       "      <td>MalevsPartheno</td>\n",
       "      <td>Male&gt;Partheno</td>\n",
       "      <td>1</td>\n",
       "      <td>6.416811e-10</td>\n",
       "      <td>10.655157</td>\n",
       "      <td>1612.581797</td>\n",
       "      <td>3.717083e-07</td>\n",
       "      <td>1855.157830</td>\n",
       "      <td>1.028444</td>\n",
       "      <td>123.374449</td>\n",
       "      <td>0.068395</td>\n",
       "      <td>B-like cysteine ase 3</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>g1052</td>\n",
       "      <td>MalevsPartheno</td>\n",
       "      <td>Male&gt;Partheno</td>\n",
       "      <td>1</td>\n",
       "      <td>3.633853e-10</td>\n",
       "      <td>10.739071</td>\n",
       "      <td>1709.158793</td>\n",
       "      <td>3.717083e-07</td>\n",
       "      <td>17499.549755</td>\n",
       "      <td>10.046700</td>\n",
       "      <td>1163.780935</td>\n",
       "      <td>0.668141</td>\n",
       "      <td>B-like cysteine ase 3</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>g13894</td>\n",
       "      <td>MalevsPartheno</td>\n",
       "      <td>Male&gt;Partheno</td>\n",
       "      <td>1</td>\n",
       "      <td>3.030168e-10</td>\n",
       "      <td>11.579582</td>\n",
       "      <td>3060.564620</td>\n",
       "      <td>3.717083e-07</td>\n",
       "      <td>4497.057113</td>\n",
       "      <td>1.369622</td>\n",
       "      <td>299.069942</td>\n",
       "      <td>0.091085</td>\n",
       "      <td>SUMO- ligase PIAS2-like</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>g13901</td>\n",
       "      <td>MalevsPartheno</td>\n",
       "      <td>Male&gt;Partheno</td>\n",
       "      <td>1</td>\n",
       "      <td>3.285241e-10</td>\n",
       "      <td>10.223023</td>\n",
       "      <td>1195.188721</td>\n",
       "      <td>3.717083e-07</td>\n",
       "      <td>4413.650911</td>\n",
       "      <td>3.534850</td>\n",
       "      <td>293.523140</td>\n",
       "      <td>0.235080</td>\n",
       "      <td>-N-acetylglucosamine--peptide N-acetylglucosam...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>g15240</td>\n",
       "      <td>MalevsPartheno</td>\n",
       "      <td>Male&gt;Partheno</td>\n",
       "      <td>1</td>\n",
       "      <td>9.454212e-10</td>\n",
       "      <td>10.325433</td>\n",
       "      <td>1283.113518</td>\n",
       "      <td>3.717083e-07</td>\n",
       "      <td>1880.041721</td>\n",
       "      <td>1.365800</td>\n",
       "      <td>125.029315</td>\n",
       "      <td>0.090830</td>\n",
       "      <td>facilitator superfamily domain-containing 6 B</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>14103</th>\n",
       "      <td>g9983</td>\n",
       "      <td>MalevsPartheno</td>\n",
       "      <td>Male=Partheno</td>\n",
       "      <td>0</td>\n",
       "      <td>9.732021e-01</td>\n",
       "      <td>0.338480</td>\n",
       "      <td>1.264423</td>\n",
       "      <td>9.999986e-01</td>\n",
       "      <td>1897.972198</td>\n",
       "      <td>1501.029867</td>\n",
       "      <td>126.221750</td>\n",
       "      <td>99.823710</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>14104</th>\n",
       "      <td>g9984</td>\n",
       "      <td>MalevsPartheno</td>\n",
       "      <td>Male=Partheno</td>\n",
       "      <td>0</td>\n",
       "      <td>9.981760e-01</td>\n",
       "      <td>0.025302</td>\n",
       "      <td>1.017693</td>\n",
       "      <td>9.999986e-01</td>\n",
       "      <td>1913.176933</td>\n",
       "      <td>1879.553424</td>\n",
       "      <td>127.232900</td>\n",
       "      <td>124.996800</td>\n",
       "      <td>catecholamines up-like</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>14105</th>\n",
       "      <td>g9985</td>\n",
       "      <td>MalevsPartheno</td>\n",
       "      <td>Male=Partheno</td>\n",
       "      <td>0</td>\n",
       "      <td>7.550673e-01</td>\n",
       "      <td>-0.593703</td>\n",
       "      <td>1.509115</td>\n",
       "      <td>9.999986e-01</td>\n",
       "      <td>143.294916</td>\n",
       "      <td>217.053953</td>\n",
       "      <td>9.529610</td>\n",
       "      <td>14.434840</td>\n",
       "      <td>ST7 homolog</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>14106</th>\n",
       "      <td>g9986</td>\n",
       "      <td>MalevsPartheno</td>\n",
       "      <td>Male=Partheno</td>\n",
       "      <td>0</td>\n",
       "      <td>6.892120e-01</td>\n",
       "      <td>0.745883</td>\n",
       "      <td>1.677001</td>\n",
       "      <td>9.999986e-01</td>\n",
       "      <td>3310.595731</td>\n",
       "      <td>1974.267320</td>\n",
       "      <td>220.166100</td>\n",
       "      <td>131.295600</td>\n",
       "      <td>pellucida domain</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>14107</th>\n",
       "      <td>g9990</td>\n",
       "      <td>MalevsPartheno</td>\n",
       "      <td>Male=Partheno</td>\n",
       "      <td>0</td>\n",
       "      <td>9.798081e-01</td>\n",
       "      <td>0.191107</td>\n",
       "      <td>1.141640</td>\n",
       "      <td>9.999986e-01</td>\n",
       "      <td>422.372894</td>\n",
       "      <td>369.858517</td>\n",
       "      <td>28.089270</td>\n",
       "      <td>24.596880</td>\n",
       "      <td>: max X</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>14108 rows × 13 columns</p>\n",
       "</div>"
      ]
     },
     "execution_count": 13,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "execution_count": 13
  },
  {
   "metadata": {
    "ExecuteTime": {
     "end_time": "2024-05-27T14:36:55.335021Z",
     "start_time": "2024-05-27T14:36:55.303279Z"
    }
   },
   "cell_type": "code",
   "source": [
    "# Keeping only the lines in the gff where there are the expressed genes\n",
    "all_expressed = all_genes_tss[(all_genes_tss['gene'].isin(all_expressed.gene))]\n",
    "# used for the run_intersects_TSS.sh\n",
    "all_expressed.to_csv('Acyrthosiphon_pisum_JIC1_v1.0_expressed_tss.bed', sep='\\t', index=False, na_rep='NA', header=False)\n",
    "all_expressed"
   ],
   "id": "e08f22c3bc34fb06",
   "outputs": [
    {
     "data": {
      "text/plain": [
       "               seqid  DEG_start  DEG_end strand    gene\n",
       "57        scaffold_1      70696    73696      +    g157\n",
       "248       scaffold_1     151776   154776      +    g166\n",
       "304       scaffold_1     182836   185836      +    g168\n",
       "338       scaffold_1     222426   225426      -    g169\n",
       "359       scaffold_1     223016   226016      +    g170\n",
       "...              ...        ...      ...    ...     ...\n",
       "892413  scaffold_508       2056     5056      -  g20117\n",
       "892434  scaffold_512       1576     4576      -  g22836\n",
       "892464  scaffold_517          0     2266      +  g20130\n",
       "892584  scaffold_527       1816     4816      +  g22813\n",
       "892632  scaffold_529          0     1501      +  g22853\n",
       "\n",
       "[14108 rows x 5 columns]"
      ],
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>seqid</th>\n",
       "      <th>DEG_start</th>\n",
       "      <th>DEG_end</th>\n",
       "      <th>strand</th>\n",
       "      <th>gene</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>57</th>\n",
       "      <td>scaffold_1</td>\n",
       "      <td>70696</td>\n",
       "      <td>73696</td>\n",
       "      <td>+</td>\n",
       "      <td>g157</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>248</th>\n",
       "      <td>scaffold_1</td>\n",
       "      <td>151776</td>\n",
       "      <td>154776</td>\n",
       "      <td>+</td>\n",
       "      <td>g166</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>304</th>\n",
       "      <td>scaffold_1</td>\n",
       "      <td>182836</td>\n",
       "      <td>185836</td>\n",
       "      <td>+</td>\n",
       "      <td>g168</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>338</th>\n",
       "      <td>scaffold_1</td>\n",
       "      <td>222426</td>\n",
       "      <td>225426</td>\n",
       "      <td>-</td>\n",
       "      <td>g169</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>359</th>\n",
       "      <td>scaffold_1</td>\n",
       "      <td>223016</td>\n",
       "      <td>226016</td>\n",
       "      <td>+</td>\n",
       "      <td>g170</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>892413</th>\n",
       "      <td>scaffold_508</td>\n",
       "      <td>2056</td>\n",
       "      <td>5056</td>\n",
       "      <td>-</td>\n",
       "      <td>g20117</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>892434</th>\n",
       "      <td>scaffold_512</td>\n",
       "      <td>1576</td>\n",
       "      <td>4576</td>\n",
       "      <td>-</td>\n",
       "      <td>g22836</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>892464</th>\n",
       "      <td>scaffold_517</td>\n",
       "      <td>0</td>\n",
       "      <td>2266</td>\n",
       "      <td>+</td>\n",
       "      <td>g20130</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>892584</th>\n",
       "      <td>scaffold_527</td>\n",
       "      <td>1816</td>\n",
       "      <td>4816</td>\n",
       "      <td>+</td>\n",
       "      <td>g22813</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>892632</th>\n",
       "      <td>scaffold_529</td>\n",
       "      <td>0</td>\n",
       "      <td>1501</td>\n",
       "      <td>+</td>\n",
       "      <td>g22853</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>14108 rows × 5 columns</p>\n",
       "</div>"
      ]
     },
     "execution_count": 14,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "execution_count": 14
  },
  {
   "cell_type": "code",
   "id": "initial_id",
   "metadata": {
    "collapsed": true,
    "ExecuteTime": {
     "end_time": "2024-05-27T14:36:58.606219Z",
     "start_time": "2024-05-27T14:36:58.584731Z"
    }
   },
   "source": [
    "# DAR + DEG broad\n",
    "DAR_broad_DEG_male_path = f'{integrated_directory}DAR_broad_DEG_male_minreps{min_reps}_overlap{overlap}.bed'\n",
    "DAR_broad_DEG_partheno_path = f'{integrated_directory}DAR_broad_DEG_partheno_minreps{min_reps}_overlap{overlap}.bed'\n",
    "DAR_broad_DEG_male = pd.read_csv(DAR_broad_DEG_male_path, sep='\\t', names=header_DAR_broad_DEG)\n",
    "DAR_broad_DEG_partheno = pd.read_csv(DAR_broad_DEG_partheno_path, sep='\\t', names=header_DAR_broad_DEG)\n",
    "\n",
    "# DAR + DEG narrow\n",
    "DAR_narrow_DEG_male_path = f'{integrated_directory}DAR_narrow_DEG_male_minreps{min_reps}_overlap{overlap}.bed'\n",
    "DAR_narrow_DEG_partheno_path = f'{integrated_directory}DAR_narrow_DEG_partheno_minreps{min_reps}_overlap{overlap}.bed'\n",
    "DAR_narrow_DEG_male = pd.read_csv(DAR_narrow_DEG_male_path, sep='\\t', names=header_DAR_narrow_DEG)\n",
    "DAR_narrow_DEG_partheno = pd.read_csv(DAR_narrow_DEG_partheno_path, sep='\\t', names=header_DAR_narrow_DEG)\n",
    "DAR_narrow_DEG_partheno"
   ],
   "outputs": [
    {
     "data": {
      "text/plain": [
       "            chr     start       end     interval_id strand    gene  num_peaks  \\\n",
       "0    scaffold_1    347008    347311     Interval_13      +    g175          1   \n",
       "1    scaffold_1    442462    442861     Interval_16      +    g180          2   \n",
       "2    scaffold_1    442184    442434     Interval_15      +    g180          1   \n",
       "3    scaffold_1   2104718   2104930     Interval_58      +    g263          1   \n",
       "4    scaffold_1  10908075  10908268    Interval_224      -    g808          1   \n",
       "..          ...       ...       ...             ...    ...     ...        ...   \n",
       "107  scaffold_4  19402907  19403140  Interval_36773      -  g21409          1   \n",
       "108  scaffold_4  19402907  19403140  Interval_36773      +  g21410          1   \n",
       "109  scaffold_4  32110557  32110772  Interval_38038      +  g22048          1   \n",
       "110  scaffold_4  34674158  34674329  Interval_38258      +  g22202          1   \n",
       "111  scaffold_4  36157176  36157360  Interval_38405      -  g22263          1   \n",
       "\n",
       "     num_samples  Male1_bool  Male2_bool  ...  Male1_end  Male2_end  \\\n",
       "0              1       False       False  ...        NaN        NaN   \n",
       "1              2       False       False  ...        NaN        NaN   \n",
       "2              1       False       False  ...        NaN        NaN   \n",
       "3              1       False       False  ...        NaN        NaN   \n",
       "4              1       False       False  ...        NaN        NaN   \n",
       "..           ...         ...         ...  ...        ...        ...   \n",
       "107            1       False       False  ...        NaN        NaN   \n",
       "108            1       False       False  ...        NaN        NaN   \n",
       "109            1       False       False  ...        NaN        NaN   \n",
       "110            1       False       False  ...        NaN        NaN   \n",
       "111            1       False       False  ...        NaN        NaN   \n",
       "\n",
       "     Partheno1_end  Partheno2_end  Partheno3_end  Male1_summit Male2_summit  \\\n",
       "0         347311.0            NaN            NaN           NaN          NaN   \n",
       "1         442861.0         442789            NaN           NaN          NaN   \n",
       "2              NaN            NaN       442434.0           NaN          NaN   \n",
       "3        2104930.0            NaN            NaN           NaN          NaN   \n",
       "4       10908268.0            NaN            NaN           NaN          NaN   \n",
       "..             ...            ...            ...           ...          ...   \n",
       "107            NaN            NaN     19403140.0           NaN          NaN   \n",
       "108            NaN            NaN     19403140.0           NaN          NaN   \n",
       "109     32110772.0            NaN            NaN           NaN          NaN   \n",
       "110            NaN            NaN     34674329.0           NaN          NaN   \n",
       "111            NaN            NaN     36157360.0           NaN          NaN   \n",
       "\n",
       "     Partheno1_summit  Partheno2_summit  Partheno3_summit  \n",
       "0                 NaN               NaN               NaN  \n",
       "1                 NaN               NaN               NaN  \n",
       "2                 NaN               NaN               NaN  \n",
       "3                 NaN               NaN               NaN  \n",
       "4                 NaN               NaN               NaN  \n",
       "..                ...               ...               ...  \n",
       "107               NaN               NaN               NaN  \n",
       "108               NaN               NaN               NaN  \n",
       "109               NaN               NaN               NaN  \n",
       "110               NaN               NaN               NaN  \n",
       "111               NaN               NaN               NaN  \n",
       "\n",
       "[112 rows x 43 columns]"
      ],
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>chr</th>\n",
       "      <th>start</th>\n",
       "      <th>end</th>\n",
       "      <th>interval_id</th>\n",
       "      <th>strand</th>\n",
       "      <th>gene</th>\n",
       "      <th>num_peaks</th>\n",
       "      <th>num_samples</th>\n",
       "      <th>Male1_bool</th>\n",
       "      <th>Male2_bool</th>\n",
       "      <th>...</th>\n",
       "      <th>Male1_end</th>\n",
       "      <th>Male2_end</th>\n",
       "      <th>Partheno1_end</th>\n",
       "      <th>Partheno2_end</th>\n",
       "      <th>Partheno3_end</th>\n",
       "      <th>Male1_summit</th>\n",
       "      <th>Male2_summit</th>\n",
       "      <th>Partheno1_summit</th>\n",
       "      <th>Partheno2_summit</th>\n",
       "      <th>Partheno3_summit</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>scaffold_1</td>\n",
       "      <td>347008</td>\n",
       "      <td>347311</td>\n",
       "      <td>Interval_13</td>\n",
       "      <td>+</td>\n",
       "      <td>g175</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>347311.0</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>scaffold_1</td>\n",
       "      <td>442462</td>\n",
       "      <td>442861</td>\n",
       "      <td>Interval_16</td>\n",
       "      <td>+</td>\n",
       "      <td>g180</td>\n",
       "      <td>2</td>\n",
       "      <td>2</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>442861.0</td>\n",
       "      <td>442789</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>scaffold_1</td>\n",
       "      <td>442184</td>\n",
       "      <td>442434</td>\n",
       "      <td>Interval_15</td>\n",
       "      <td>+</td>\n",
       "      <td>g180</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>442434.0</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>scaffold_1</td>\n",
       "      <td>2104718</td>\n",
       "      <td>2104930</td>\n",
       "      <td>Interval_58</td>\n",
       "      <td>+</td>\n",
       "      <td>g263</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>2104930.0</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>scaffold_1</td>\n",
       "      <td>10908075</td>\n",
       "      <td>10908268</td>\n",
       "      <td>Interval_224</td>\n",
       "      <td>-</td>\n",
       "      <td>g808</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>10908268.0</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>107</th>\n",
       "      <td>scaffold_4</td>\n",
       "      <td>19402907</td>\n",
       "      <td>19403140</td>\n",
       "      <td>Interval_36773</td>\n",
       "      <td>-</td>\n",
       "      <td>g21409</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>19403140.0</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>108</th>\n",
       "      <td>scaffold_4</td>\n",
       "      <td>19402907</td>\n",
       "      <td>19403140</td>\n",
       "      <td>Interval_36773</td>\n",
       "      <td>+</td>\n",
       "      <td>g21410</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>19403140.0</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>109</th>\n",
       "      <td>scaffold_4</td>\n",
       "      <td>32110557</td>\n",
       "      <td>32110772</td>\n",
       "      <td>Interval_38038</td>\n",
       "      <td>+</td>\n",
       "      <td>g22048</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>32110772.0</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>110</th>\n",
       "      <td>scaffold_4</td>\n",
       "      <td>34674158</td>\n",
       "      <td>34674329</td>\n",
       "      <td>Interval_38258</td>\n",
       "      <td>+</td>\n",
       "      <td>g22202</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>34674329.0</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>111</th>\n",
       "      <td>scaffold_4</td>\n",
       "      <td>36157176</td>\n",
       "      <td>36157360</td>\n",
       "      <td>Interval_38405</td>\n",
       "      <td>-</td>\n",
       "      <td>g22263</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>36157360.0</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>112 rows × 43 columns</p>\n",
       "</div>"
      ]
     },
     "execution_count": 15,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "execution_count": 15
  },
  {
   "metadata": {
    "ExecuteTime": {
     "end_time": "2024-05-27T14:37:00.647347Z",
     "start_time": "2024-05-27T14:37:00.614593Z"
    }
   },
   "cell_type": "code",
   "source": [
    "# DAR in TSS regions (broad)\n",
    "DAR_broad_TSS_male_path = f'{integrated_directory}DAR_broad_TSS_male_minreps{min_reps}_overlap{overlap}.bed'\n",
    "DAR_broad_TSS_partheno_path = f'{integrated_directory}DAR_broad_TSS_partheno_minreps{min_reps}_overlap{overlap}.bed'\n",
    "DAR_broad_TSS_male = pd.read_csv(DAR_broad_TSS_male_path, sep='\\t', names=header_DAR_broad_DEG)\n",
    "DAR_broad_TSS_partheno = pd.read_csv(DAR_broad_TSS_partheno_path, sep='\\t', names=header_DAR_broad_DEG)\n",
    "\n",
    "# DAR in TSS regions (narrow)\n",
    "DAR_narrow_TSS_male_path = f'{integrated_directory}DAR_narrow_TSS_male_minreps{min_reps}_overlap{overlap}.bed'\n",
    "DAR_narrow_TSS_partheno_path = f'{integrated_directory}DAR_narrow_TSS_partheno_minreps{min_reps}_overlap{overlap}.bed'\n",
    "DAR_narrow_TSS_male = pd.read_csv(DAR_narrow_TSS_male_path, sep='\\t', names=header_DAR_narrow_DEG)\n",
    "DAR_narrow_TSS_partheno = pd.read_csv(DAR_narrow_TSS_partheno_path, sep='\\t', names=header_DAR_narrow_DEG)\n",
    "\n",
    "DAR_broad_TSS_male"
   ],
   "id": "8d79d7300e01eab6",
   "outputs": [
    {
     "data": {
      "text/plain": [
       "               chr    start      end     interval_id strand    gene  \\\n",
       "0       scaffold_1  1054779  1055762     Interval_46      -    g207   \n",
       "1       scaffold_1  2447629  2447933     Interval_96      -    g283   \n",
       "2       scaffold_1  8508176  8508404    Interval_269      -    g692   \n",
       "3       scaffold_1  8512688  8513408    Interval_270      +    g693   \n",
       "4       scaffold_1  9375101  9375387    Interval_298      -    g728   \n",
       "...            ...      ...      ...             ...    ...     ...   \n",
       "1916  scaffold_142    12301    12875   Interval_9986      +  g22896   \n",
       "1917  scaffold_152      691     1979   Interval_9995      +    g146   \n",
       "1918  scaffold_183     9853    10433  Interval_10010      +  g22851   \n",
       "1919  scaffold_263     7453     7925  Interval_27260      -  g22722   \n",
       "1920  scaffold_446     2887     3439  Interval_44171      -  g13015   \n",
       "\n",
       "      num_peaks  num_samples  Male1_bool  Male2_bool  ...  Male1_start  \\\n",
       "0             1            1       False        True  ...          NaN   \n",
       "1             1            1       False        True  ...          NaN   \n",
       "2             1            1       False        True  ...          NaN   \n",
       "3             1            1       False        True  ...          NaN   \n",
       "4             1            1        True       False  ...      9375101   \n",
       "...         ...          ...         ...         ...  ...          ...   \n",
       "1916          1            1       False        True  ...          NaN   \n",
       "1917          1            1       False        True  ...          NaN   \n",
       "1918          1            1       False        True  ...          NaN   \n",
       "1919          1            1       False        True  ...          NaN   \n",
       "1920          1            1       False        True  ...          NaN   \n",
       "\n",
       "      Male2_start  Partheno1_start Partheno2_start  Partheno3_start  \\\n",
       "0       1054779.0              NaN             NaN              NaN   \n",
       "1       2447629.0              NaN             NaN              NaN   \n",
       "2       8508176.0              NaN             NaN              NaN   \n",
       "3       8512688.0              NaN             NaN              NaN   \n",
       "4             NaN              NaN             NaN              NaN   \n",
       "...           ...              ...             ...              ...   \n",
       "1916      12301.0              NaN             NaN              NaN   \n",
       "1917        691.0              NaN             NaN              NaN   \n",
       "1918       9853.0              NaN             NaN              NaN   \n",
       "1919       7453.0              NaN             NaN              NaN   \n",
       "1920       2887.0              NaN             NaN              NaN   \n",
       "\n",
       "      Male1_end  Male2_end  Partheno1_end Partheno2_end  Partheno3_end  \n",
       "0           NaN  1055762.0            NaN           NaN            NaN  \n",
       "1           NaN  2447933.0            NaN           NaN            NaN  \n",
       "2           NaN  8508404.0            NaN           NaN            NaN  \n",
       "3           NaN  8513408.0            NaN           NaN            NaN  \n",
       "4       9375387        NaN            NaN           NaN            NaN  \n",
       "...         ...        ...            ...           ...            ...  \n",
       "1916        NaN    12875.0            NaN           NaN            NaN  \n",
       "1917        NaN     1979.0            NaN           NaN            NaN  \n",
       "1918        NaN    10433.0            NaN           NaN            NaN  \n",
       "1919        NaN     7925.0            NaN           NaN            NaN  \n",
       "1920        NaN     3439.0            NaN           NaN            NaN  \n",
       "\n",
       "[1921 rows x 38 columns]"
      ],
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>chr</th>\n",
       "      <th>start</th>\n",
       "      <th>end</th>\n",
       "      <th>interval_id</th>\n",
       "      <th>strand</th>\n",
       "      <th>gene</th>\n",
       "      <th>num_peaks</th>\n",
       "      <th>num_samples</th>\n",
       "      <th>Male1_bool</th>\n",
       "      <th>Male2_bool</th>\n",
       "      <th>...</th>\n",
       "      <th>Male1_start</th>\n",
       "      <th>Male2_start</th>\n",
       "      <th>Partheno1_start</th>\n",
       "      <th>Partheno2_start</th>\n",
       "      <th>Partheno3_start</th>\n",
       "      <th>Male1_end</th>\n",
       "      <th>Male2_end</th>\n",
       "      <th>Partheno1_end</th>\n",
       "      <th>Partheno2_end</th>\n",
       "      <th>Partheno3_end</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>scaffold_1</td>\n",
       "      <td>1054779</td>\n",
       "      <td>1055762</td>\n",
       "      <td>Interval_46</td>\n",
       "      <td>-</td>\n",
       "      <td>g207</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>False</td>\n",
       "      <td>True</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>1054779.0</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>1055762.0</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>scaffold_1</td>\n",
       "      <td>2447629</td>\n",
       "      <td>2447933</td>\n",
       "      <td>Interval_96</td>\n",
       "      <td>-</td>\n",
       "      <td>g283</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>False</td>\n",
       "      <td>True</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>2447629.0</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>2447933.0</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>scaffold_1</td>\n",
       "      <td>8508176</td>\n",
       "      <td>8508404</td>\n",
       "      <td>Interval_269</td>\n",
       "      <td>-</td>\n",
       "      <td>g692</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>False</td>\n",
       "      <td>True</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>8508176.0</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>8508404.0</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>scaffold_1</td>\n",
       "      <td>8512688</td>\n",
       "      <td>8513408</td>\n",
       "      <td>Interval_270</td>\n",
       "      <td>+</td>\n",
       "      <td>g693</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>False</td>\n",
       "      <td>True</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>8512688.0</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>8513408.0</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>scaffold_1</td>\n",
       "      <td>9375101</td>\n",
       "      <td>9375387</td>\n",
       "      <td>Interval_298</td>\n",
       "      <td>-</td>\n",
       "      <td>g728</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>True</td>\n",
       "      <td>False</td>\n",
       "      <td>...</td>\n",
       "      <td>9375101</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>9375387</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1916</th>\n",
       "      <td>scaffold_142</td>\n",
       "      <td>12301</td>\n",
       "      <td>12875</td>\n",
       "      <td>Interval_9986</td>\n",
       "      <td>+</td>\n",
       "      <td>g22896</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>False</td>\n",
       "      <td>True</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>12301.0</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>12875.0</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1917</th>\n",
       "      <td>scaffold_152</td>\n",
       "      <td>691</td>\n",
       "      <td>1979</td>\n",
       "      <td>Interval_9995</td>\n",
       "      <td>+</td>\n",
       "      <td>g146</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>False</td>\n",
       "      <td>True</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>691.0</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>1979.0</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1918</th>\n",
       "      <td>scaffold_183</td>\n",
       "      <td>9853</td>\n",
       "      <td>10433</td>\n",
       "      <td>Interval_10010</td>\n",
       "      <td>+</td>\n",
       "      <td>g22851</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>False</td>\n",
       "      <td>True</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>9853.0</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>10433.0</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1919</th>\n",
       "      <td>scaffold_263</td>\n",
       "      <td>7453</td>\n",
       "      <td>7925</td>\n",
       "      <td>Interval_27260</td>\n",
       "      <td>-</td>\n",
       "      <td>g22722</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>False</td>\n",
       "      <td>True</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>7453.0</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>7925.0</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1920</th>\n",
       "      <td>scaffold_446</td>\n",
       "      <td>2887</td>\n",
       "      <td>3439</td>\n",
       "      <td>Interval_44171</td>\n",
       "      <td>-</td>\n",
       "      <td>g13015</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>False</td>\n",
       "      <td>True</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>2887.0</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>3439.0</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>1921 rows × 38 columns</p>\n",
       "</div>"
      ]
     },
     "execution_count": 16,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "execution_count": 16
  },
  {
   "metadata": {},
   "cell_type": "markdown",
   "source": [
    "# Chi-squared statistical test (TO CORRECT)\n",
    "## Step 1: Create a contingency table\n",
    "### Males \n",
    "| MALE    | peak (DAR)    \t                             | no peak (DAR)                                  |\n",
    "|---------|---------------------------------------------|------------------------------------------------|\n",
    "| DEG     | gene differentially expressed with peak     | gene differentially expressed without peak     |\n",
    "| notDEG  | gene not differentially expressed with peak | gene not differentially expressed without peak |\n",
    "\n",
    "-> Goal : answering the question \"Is there a statistically significant association between the differential expression status of genes (DEG vs. not DEG) and the presence of a peak in differential accessible regions (DARs) within each sex?\" \n"
   ],
   "id": "b27940e1b96634e8"
  },
  {
   "metadata": {
    "ExecuteTime": {
     "end_time": "2024-05-27T14:37:04.677248Z",
     "start_time": "2024-05-27T14:37:04.673348Z"
    }
   },
   "cell_type": "code",
   "source": [
    "# broad\n",
    "DAR_broad_DEG_male_count = DAR_broad_DEG_male.shape[0]\n",
    "notDAR_broad_DEG_male_count = DEG_male_tss_bed.shape[0] - DAR_broad_DEG_male.shape[0]\n",
    "DAR_broad_notDEG_male_count = DAR_broad_TSS_male.shape[0] - DAR_broad_DEG_male.shape[0]\n",
    "notDAR_broad_notDEG_male_count = all_expressed.shape[0] - DAR_broad_DEG_male_count - notDAR_broad_DEG_male_count - DAR_broad_notDEG_male_count\n",
    "\n",
    "# contingency table for broad DAR male\n",
    "table_broad_male = np.array([[DAR_broad_DEG_male_count, notDAR_broad_DEG_male_count],\n",
    "                  [DAR_broad_notDEG_male_count, notDAR_broad_notDEG_male_count]])\n",
    "\n",
    "table_broad_male"
   ],
   "id": "757d79e4d43948b2",
   "outputs": [
    {
     "data": {
      "text/plain": [
       "array([[ 329, 2538],\n",
       "       [1592, 9649]])"
      ]
     },
     "execution_count": 17,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "execution_count": 17
  },
  {
   "metadata": {
    "ExecuteTime": {
     "end_time": "2024-05-27T14:37:05.381097Z",
     "start_time": "2024-05-27T14:37:05.377467Z"
    }
   },
   "cell_type": "code",
   "source": [
    "# narrow\n",
    "DAR_narrow_DEG_male_count = DAR_narrow_DEG_male.shape[0]\n",
    "notDAR_narrow_DEG_male_count = DEG_male_tss_bed.shape[0] - DAR_narrow_DEG_male.shape[0]\n",
    "DAR_narrow_notDEG_male_count = DAR_narrow_TSS_male.shape[0] - DAR_narrow_DEG_male.shape[0]\n",
    "notDAR_narrow_notDEG_male_count = all_expressed.shape[0] - DAR_narrow_DEG_male_count - notDAR_narrow_DEG_male_count - DAR_narrow_notDEG_male_count\n",
    "\n",
    "# contingency table for narrow DAR male\n",
    "table_narrow_male = np.array([[DAR_narrow_DEG_male_count, notDAR_narrow_DEG_male_count],\n",
    "                  [DAR_narrow_notDEG_male_count, notDAR_narrow_notDEG_male_count]])\n",
    "\n",
    "table_narrow_male"
   ],
   "id": "26474a100da1190f",
   "outputs": [
    {
     "data": {
      "text/plain": [
       "array([[ 396, 2471],\n",
       "       [2018, 9223]])"
      ]
     },
     "execution_count": 18,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "execution_count": 18
  },
  {
   "metadata": {},
   "cell_type": "markdown",
   "source": [
    "### Parthenos\n",
    "One per sex, so for parthenos also:\n",
    "\n",
    "| PARTHENO | peak (DAR)    \t                              | no peak (DAR)                                  |\n",
    "|----------|----------------------------------------------|------------------------------------------------|\n",
    "| DEG      | gene differentially expressed with peak      | gene differentially expressed without peak     |\n",
    "| notDEG   | gene not differentially expressed with peak  | gene not differentially expressed without peak |"
   ],
   "id": "d4f3c77237a09d95"
  },
  {
   "metadata": {
    "ExecuteTime": {
     "end_time": "2024-05-27T14:37:06.799600Z",
     "start_time": "2024-05-27T14:37:06.795953Z"
    }
   },
   "cell_type": "code",
   "source": [
    "# broad\n",
    "DAR_broad_DEG_partheno_count = DAR_broad_DEG_partheno.shape[0]\n",
    "notDAR_broad_DEG_partheno_count = DEG_partheno_tss_bed.shape[0] - DAR_broad_DEG_partheno.shape[0]\n",
    "DAR_broad_notDEG_partheno_count = DAR_broad_TSS_partheno.shape[0] - DAR_broad_DEG_partheno.shape[0]\n",
    "notDAR_broad_notDEG_partheno_count = all_expressed.shape[0] - DAR_broad_DEG_partheno_count - notDAR_broad_DEG_partheno_count - DAR_broad_notDEG_partheno_count\n",
    "\n",
    "# contingency table for broad DAR partheno\n",
    "table_broad_partheno = np.array([[DAR_broad_DEG_partheno_count, notDAR_broad_DEG_partheno_count],\n",
    "                  [DAR_broad_notDEG_partheno_count, notDAR_broad_notDEG_partheno_count]])\n",
    "\n",
    "table_broad_partheno"
   ],
   "id": "ea59835b8fbea794",
   "outputs": [
    {
     "data": {
      "text/plain": [
       "array([[   93,  1716],\n",
       "       [  424, 11875]])"
      ]
     },
     "execution_count": 19,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "execution_count": 19
  },
  {
   "metadata": {
    "ExecuteTime": {
     "end_time": "2024-05-27T14:37:07.507323Z",
     "start_time": "2024-05-27T14:37:07.503435Z"
    }
   },
   "cell_type": "code",
   "source": [
    "# narrow\n",
    "DAR_narrow_DEG_partheno_count = DAR_narrow_DEG_partheno.shape[0]\n",
    "notDAR_narrow_DEG_partheno_count = DEG_partheno_tss_bed.shape[0] - DAR_narrow_DEG_partheno.shape[0]\n",
    "DAR_narrow_notDEG_partheno_count = DAR_narrow_TSS_partheno.shape[0] - DAR_narrow_DEG_partheno.shape[0]\n",
    "notDAR_narrow_notDEG_partheno_count = all_expressed.shape[0] - DAR_narrow_DEG_partheno_count - notDAR_narrow_DEG_partheno_count - DAR_narrow_notDEG_partheno_count\n",
    "\n",
    "# contingency table for narrow DAR partheno\n",
    "table_narrow_partheno = np.array([[DAR_narrow_DEG_partheno_count, notDAR_narrow_DEG_partheno_count],\n",
    "                  [DAR_narrow_notDEG_partheno_count, notDAR_narrow_notDEG_partheno_count]])\n",
    "\n",
    "table_narrow_partheno"
   ],
   "id": "da9ceb8c5ce01288",
   "outputs": [
    {
     "data": {
      "text/plain": [
       "array([[  112,  1697],\n",
       "       [  443, 11856]])"
      ]
     },
     "execution_count": 20,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "execution_count": 20
  },
  {
   "metadata": {},
   "cell_type": "markdown",
   "source": "## Step 2 : Apply the chi-squared test on that crosstab",
   "id": "935b7ffb1ac1f48e"
  },
  {
   "metadata": {
    "ExecuteTime": {
     "end_time": "2024-05-27T14:37:09.547251Z",
     "start_time": "2024-05-27T14:37:09.542510Z"
    }
   },
   "cell_type": "code",
   "source": [
    "crosstabs = [\n",
    "    (table_broad_male, 'Broad male'),\n",
    "    (table_broad_partheno, 'Broad partheno'),\n",
    "    (table_narrow_male, 'Narrow male'),\n",
    "    (table_narrow_partheno, 'Narrow partheno')\n",
    "]\n",
    "\n",
    "for table, label in crosstabs:\n",
    "    chi2, p_value, dof, expected = chi2_contingency(table)\n",
    "    \n",
    "    print(label)\n",
    "    print('Chi-squared Statistic:', chi2)\n",
    "    print('P-value:', p_value)\n",
    "    print('Expected Frequencies:', expected)\n",
    "    print('\\n')"
   ],
   "id": "c3ddb48ca4c8710f",
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Broad male\n",
      "Chi-squared Statistic: 13.794789454257891\n",
      "P-value: 0.00020390094405709614\n",
      "Expected Frequencies: [[ 390.38184009 2476.61815991]\n",
      " [1530.61815991 9710.38184009]]\n",
      "\n",
      "\n",
      "Broad partheno\n",
      "Chi-squared Statistic: 12.336755375664117\n",
      "P-value: 0.0004441262815130554\n",
      "Expected Frequencies: [[   66.2923873  1742.7076127]\n",
      " [  450.7076127 11848.2923873]]\n",
      "\n",
      "\n",
      "Narrow male\n",
      "Chi-squared Statistic: 27.311793822433486\n",
      "P-value: 1.7315206713409603e-07\n",
      "Expected Frequencies: [[ 490.56833003 2376.43166997]\n",
      " [1923.43166997 9317.56833003]]\n",
      "\n",
      "\n",
      "Narrow partheno\n",
      "Chi-squared Statistic: 27.29762965123209\n",
      "P-value: 1.7442510286788888e-07\n",
      "Expected Frequencies: [[   71.16494188  1737.83505812]\n",
      " [  483.83505812 11815.16494188]]\n",
      "\n",
      "\n"
     ]
    }
   ],
   "execution_count": 21
  },
  {
   "metadata": {},
   "cell_type": "markdown",
   "source": [
    "- Chi-squared statistic: measures the discrepancy between the observed and expected frequencies in the crosstab -> very low because our value are quite similar between observed and expected ?\n",
    "- p-value not < 0.05: cannot reject the null hypothesis -> no statistically significant association between the expression and accessibility across male and parthenogenetic samples\n",
    "- Expected Frequencies for the males: 44.8 for DAR_narrow_DEG_male_count and 2822.2 for the DEGs not DARs (the DEG_male_tss_bed that are not in the DEG + DARs list. For the parthenos: 28.2 and 1780.6."
   ],
   "id": "f37a83f9d996ac35"
  },
  {
   "metadata": {},
   "cell_type": "markdown",
   "source": "",
   "id": "156104c49641e0a8"
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "Python 3",
   "language": "python",
   "name": "python3"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 2
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython2",
   "version": "2.7.6"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 5
}
